src/Algebra/Graph/Undirected.hs

```diff
@@ -0,0 +1,870 @@
+{-# LANGUAGE DeriveGeneric, RankNTypes #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Undirected
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for the
+-- motivation behind the library, the underlying theory, and implementation details.
+--
+-- This module defines and abstraction over the core data type 'Graph' and associated algorithms.
+-- Undirected graphs satisfy all laws of the
+-- 'Algebra.Graph.Class.Undirected' type class, including the commutativity
+-- of 'connect'.
+-- For graphs that are known to be /non-empty/ at compile time, see
+-- "Algebra.Graph.NonEmpty". 'Graph' is an instance of type classes defined in
```

Comment:
The comment about non-empty graphs is out-of-date unless you plan to implement non-empty undirected graphs too. I suggest to just delete it for now.

Reply:
Yes, my bad!